Scripts/Editor/Drawers/Odin/AsDynamicPortAttributes.cs

```diff
@@ -0,0 +1,98 @@
+#if UNITY_EDITOR && ODIN_INSPECTOR
+using Sirenix.OdinInspector.Editor;
+using System;
+using UnityEditor;
+using UnityEngine;
+using XNode;
+using static XNode.Node;
+
+namespace XNodeEditor.Odin
+{
+	internal abstract class AsDynamicPortAtribute : System.Attribute
```

Comment:
What is this attribute for?

Reply:
They're attached dynamically by the collection resolver so that it knows to draw them with those drawers. They're not intended for external use.

Reply:
I am also technically abusing attributes to pass the port to the drawer.

Reply:
Alright. Was afraid this was something you expected the user to decorate his fields with.

Reply:
I'm very particular about things like that when it's not just for my use. I could probably mark every one of the Odin structures as internal as none of them should ever be required. The way this is set up users should only have to create their own drawers / attributes as per usual Odin ways - but you never know what and end user might want to do.